openquake/hazardlib/gsim/nshmp_2014.py

```diff
@@ -111,94 +133,38 @@ def get_weighted_poes(gsim, sctx, rctx, dctx, imt, imls, truncation_level,
         Weightings as a list of tuples of (weight, number standard deviations
         of the epistemic uncertainty adjustment)
     """
-    if truncation_level is not None and truncation_level < 0:
-        raise ValueError('truncation level must be zero, positive number '
-                         'or None')
-    gsim._check_imt(imt)
-    adjustment = nga_west2_epistemic_adjustment(rctx.mag, dctx.rrup)
-    adjustment = adjustment.reshape(adjustment.shape + (1, ))
-    if truncation_level == 0:
-        # zero truncation mode, just compare imls to mean
-        imls = gsim.to_distribution_values(imls)
-        mean, _ = gsim.get_mean_and_stddevs(sctx, rctx, dctx, imt, [])
-        mean = mean.reshape(mean.shape + (1, ))
-        output = np.zeros([mean.shape[0], imls.shape[0]])
-        for (wgt, fct) in weighting:
-            output += (wgt *
-                       (imls <= (mean + (fct * adjustment))).astype(float))
-        return output
-    else:
-        # use real normal distribution
-        assert (const.StdDev.TOTAL
-                in gsim.DEFINED_FOR_STANDARD_DEVIATION_TYPES)
-        imls = gsim.to_distribution_values(imls)
-        mean, [stddev] = gsim.get_mean_and_stddevs(sctx, rctx, dctx, imt,
-                                                   [const.StdDev.TOTAL])
-        mean = mean.reshape(mean.shape + (1, ))
-        stddev = stddev.reshape(stddev.shape + (1, ))
-        output = np.zeros([mean.shape[0], imls.shape[0]])
-        for (wgt, fct) in weighting:
-            values = (imls - (mean + (fct * adjustment))) / stddev
-            if truncation_level is None:
-                output += (wgt * _norm_sf(values))
-            else:
-                output += (wgt * _truncnorm_sf(truncation_level, values))
-        return output
+    mean = np.array(mean_std[0])  # make a copy
+    output = np.zeros([len(mean), len(imls)])
+    for w, s in weighting:
+        mean_std[0] = mean + s * gsim.adjustment
+        output += gsim.__class__.__base__.get_poes(
+            gsim, mean_std, imls, truncation_level) * w
+    return output
 
 
 class AbrahamsonEtAl2014NSHMPUpper(AbrahamsonEtAl2014):
     """
     Implements the positive NSHMP adjustment factor for the Abrahamson et al.
     (2014) NGA West 2 GMPE
     """
-    def get_mean_and_stddevs(self, sctx, rctx, dctx, imt, stddev_types):
-        """
-        See :meth:`superclass method
-        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
-        for spec of input and result values.
-        """
-        # Get original mean and standard deviations
-        mean, stddevs = super().get_mean_and_stddevs(
-            sctx, rctx, dctx, imt, stddev_types)
-        # Return mean, increased by the adjustment factor,
-        # and standard devation
-        return mean + nga_west2_epistemic_adjustment(rctx.mag, dctx.rrup),\
-            stddevs
+    get_mean_and_stddevs = get_mean_and_stddevs
```

Comment:
I do not understand this.

Reply:
This is used to convert a function external to the class into a method of the class